scripts/build_clone_docs.sh

```diff
@@ -1,25 +1,60 @@
 # Clone the docs (draft or published)
 #
 # git clone does not like to clone into folders that are populated.  We are doing
-# this sequence of commands to workaround that limitation. 
+# this sequence of commands to workaround that limitation.
 # Could _not_ use:
 #   git clone git@github.com:kabanero-io/docs.git --branch develop src/main/content
 DOCS_GIT_URL=${1:-"https://github.com/kabanero-io/docs"}
 DOCS_GIT_REVISION=${2:-"master"}
 
 CUR_DIR="$(cd $(dirname $0) && pwd)"
+
+git_clone_doc_tags() {
+    #get all tags from the repo
+    TAGS=$(git ls-remote --tags $1 | cut -d/ -f3-)
+    #get the latest tag from repo
+    LATEST_TAG=$(git ls-remote --tags $DOCS_GIT_URL | awk -F'/' '/[0-9].[0-9].[0-9].*/ { print $3}' | sort -nr | head -n1)
+    
+    for TAG in $TAGS; do
+    #Remove the letter v or any other letter before the version number so VERSION only contains the numaric value of a release
+    VERSION=$(sed 's/[A-Za-z]*//g' <<<$TAG)
+
+    # If its the latest version of docs we want to clone it into the root of /docs
+    # for all other tags we clone them into a folder of thier own with the folder name being the version of doc under the /docs dir 
+        if [ "$TAG" == "$LATEST_TAG" ]; then
+            git clone -b $TAG --single-branch $DOCS_GIT_URL .
+
+            #Create an adoc file in the root of the docs folder and add the list of tagged version for later use
+            echo ":page-title: docs version list" >> docs-version-list.adoc 
```

Comment:
a cleaner way might be to create a docversions.json file and put the versions as an array in there. Then on the client side you can do a ajax request for docversions.json which will return that array for you.

It might help clean up some of the liquid you have on the front end. Since all the paths already exist all you would need to do is have some js that creates the drop down and changes the URL

Reply:
so you can play around with the below code in the docs layout file. That would get you all docs with 0.0.4 in the URL (so all docs under the 0.0.4 directory) You'll need to find a way to variablize the 0.0.4 in the example but it might be possible with parsing the page.url for the current page since it should have the version in it, for example `/docs/0.0.4/......`

```
            {% for page in site.pages | where "layout", "doc" %}            
                {% if page.url contains "/docs/0.0.4" %}
                    {{page.url}}
                {% endif %}
            {% endfor %}
```